File: src/test/java/eu/bitwalker/useragentutils/browser/LotusParameterizedTest.java
package eu.bitwalker.useragentutils.browser;

import static java.util.Arrays.asList;

import java.util.Collection;

import org.junit.runners.Parameterized.Parameters;

import eu.bitwalker.useragentutils.Browser;
import eu.bitwalker.useragentutils.OperatingSystem;

public class LotusParameterizedTest extends AbstractUserAgentParameterizedTest {

	public LotusParameterizedTest(String userAgentValue,
			Browser expectedBrowser, String browserVersionExpected,
			OperatingSystem osExpected) {
		super(userAgentValue, expectedBrowser, browserVersionExpected,
				osExpected);
	}

	@Parameters
	public static Collection<Object[]> testData() {
		return asList(new Object[][] {
				// lotusNotes
				{ "Mozilla/4.0 (compatible; Lotus-Notes/5.0; Windows-NT)",
						Browser.LOTUS_NOTES, "5.0", OperatingSystem.WINDOWS },
				{ "Mozilla/4.0 (compatible; Lotus-Notes/6.0; Windows-NT)",
						Browser.LOTUS_NOTES, "6.0", OperatingSystem.WINDOWS } 
		});
	}
}

File: src/test/csharp/eu/bitwalker/useragentutils/browser/LotusParameterizedTest.cs
using System;
using NUnit.Framework;

namespace  eu.bitwalker.useragentutils.browser {


public class LotusParameterizedTest : AbstractUserAgentParameterizedTest {
				// lotusNotes
				[TestCase( "Mozilla/4.0 (compatible; Lotus-Notes/5.0; Windows-NT)",
						"LOTUS_NOTES", "5.0", "WINDOWS" )]
				[TestCase( "Mozilla/4.0 (compatible; Lotus-Notes/6.0; Windows-NT)",
						"LOTUS_NOTES", "6.0", "WINDOWS" )]
    [Test]
    public void testData(String userAgentValue,
            string expectedBrowserStr, String expectedBrowserVersion,
            string expectedOSStr)
    {
        shouldParseUserAgent(userAgentValue, Browser.valueOf(expectedBrowserStr), expectedBrowserVersion, OperatingSystem.valueOf(expectedOSStr));
    }
}
}
